scheduler.go

```diff
@@ -362,6 +404,9 @@ func (s *Scheduler) StartImmediately() *Scheduler {
 
 // shouldRun returns true if the Job should be run now
 func (s *Scheduler) shouldRun(j *Job) bool {
+	if j.neverRan() && j.startsImmediately {
```

Comment:
perhaps could avoid calling neverRan by setting startsImmediately to false always after first reschedule; currently we do it on weekday scheduling